data_steward/tools/participant_prevalence.py

```diff
@@ -0,0 +1,244 @@
+        bq_pid_str = str(pid_list)[1:-1]
+        logging.info(f"Generated BQ pid string: {bq_pid_str}")
+        return bq_pid_str
+    elif pid_project_id and sandbox_dataset_id and pid_table_id:
+        pid_query = consts.PID_QUERY.format(pid_project=pid_project_id,
+                                            sandbox_dataset=sandbox_dataset_id,
+                                            pid_table=pid_table_id)
+        logging.info(f"Generated pid query: {pid_query}")
+        return pid_query
+    else:
+        raise ValueError('Please specify pids or pid_table parameters')
+
+
+def get_cdm_tables_with_person_id(project_id, dataset_id):
+    """
+    Get df of table_ids(first column) and table names that have a person_id column as the second column
+
+    :param project_id: identifies the project
+    :param dataset_id: identifies the dataset
+    :return list containing tables
+    """
+    person_table_query = consts.PERSON_TABLE_QUERY.format(project=project_id,
+                                                          dataset=dataset_id)
+    result_df = bq.query(person_table_query)
+    tables_list = result_df.get(consts.TABLE_NAME_COLUMN).to_list()
+    return tables_list
+
+
+def get_pid_counts_query(project_id, dataset_id, hpo_id, pids_string, for_cdm):
+    """
+    Returns query to fetch table_name, all_counts and ehr_counts of rows pertaining to the participants
+
+    :param project_id: identifies the project
+    :param dataset_id: identifies the dataset
+    :param hpo_id: identifies the hpo site that submitted the pids
+    :param pids_string: string containing pids or pid_query
+    :param for_cdm: Boolean indicating if counting only cdm tables (excl. person)
+    :return: query selecting table_name, all_counts and ehr_counts
+    """
+    cdm_person_tables = get_cdm_tables_with_person_id(project_id, dataset_id)
+    all_person_tables = srp.get_tables_with_person_id(project_id, dataset_id)
+    if for_cdm:
+        # cdm_person_tables does not include person, but includes
+        # cdm tables where the pid column is in ordinal position 2
+        person_tables = cdm_person_tables
+    else:
+        # consider all tables with pid column but exclude the ones considered above
+        person_tables = list(set(all_person_tables) - set(cdm_person_tables))
+    pid_query_list = []
+    for table in person_tables:
+        count_types = consts.SELECT_ALL_COUNT + ',' + consts.SELECT_ZERO_COUNT
+        if for_cdm:
+            count_types = consts.SELECT_ALL_COUNT + ',' + consts.SELECT_EHR_COUNT.format(
+                table_id=table + '_id',
+                const=common.ID_CONSTANT_FACTOR + common.RDR_ID_CONSTANT)
+        pid_table_query = consts.PARTICIPANT_ROWS.format(
+            project=project_id,
+            dataset=dataset_id,
+            table=table,
+            count_types=count_types,
+            pids_string=pids_string)
+        pid_query_list.append(pid_table_query)
+    if len(pid_query_list) > 20:
```

Comment:
Sorry, I'm not sure what is happening here, please explain over phone?

Reply:
Please add a comment saying this is to avoid query length limit.

Reply:
If it is an EHR dataset then we will only get the HPO-specific tables, otherwise we get all the tables.